code/game/objects/items.dm

```diff
@@ -172,6 +172,35 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
 
 	var/canMouseDown = FALSE
 
+//modification functionality//
+	var/mods = 0
+
+	//maximum number of modifications an item can have.
+	var/maxmodify = 3
+	var/postfixed = FALSE//prevents more then one postfix. Prevents "burning sword of the jester of the dervish" and instead causes "burning prank sword of the dervish", see the naming part of sharpener code.
+	//weapon poison effect, injected on every stab if poisoned is TRUE. Amount is determined by poison power.
+	var/poisoned = FALSE
+	var/poisonpower = 1
+	var/stabpoison = /datum/reagent/toxin
+	//weapon flame effect. Ignites and burns based on burnpower if burner is TRUE
+	var/burner = FALSE
+	var/burnpower = 1
+	//weapon chill effect. Reduces temp, slightly burns based on freezepower. Applies a slowly-increasing status effect that eventually freezes in a block of ice.
+	var/freezer = FALSE
+	var/freezepower = 1
+	//finally, (semi)functional modular possession. When haunted is true, alt-click the item in question to summon a spirit, a la. the talking sword.
+	var/haunted = FALSE
+	var/possessed = FALSE
+	//modularizes bleed effect. Bleedpower dictates how quickly stacks are gained on attack.
+	var/serrated = FALSE
+	var/bleedpower = 1
+
+//attack speed modification: decreases cooldown between clicks. Lower numbers are faster, higher are slower//
+	var/attackspeed = 1
+
+//causes an effect overlay to appear over your char when the item is held inhand. Most often used for "shielding" effects, see red and blue holy staves for an example.
+	var/overlayed = FALSE
+	var/effect_icon = null//pathing is under icons/effects/effects.dmi
```

Comment:
I think some of these booleans should be relegated to item flags, if possible.

Reply:
I'll investigate doing this as part of the eris item mod port